astropy/visualization/wcsaxes/tests/test_misc.py

```diff
@@ -446,3 +447,43 @@ def test_time_wcs(time_spectral_wcs_2d):
     # with a time axis.
 
     plt.subplot(projection=time_spectral_wcs_2d)
+
+
+@ignore_matplotlibrc
+def test_simplify_labels_usetex(tmpdir):
+    """Regression test for https://github.com/astropy/astropy/issues/8004."""
+    # pytest.mark.skipif() did not work as expected, so we do this here.
+    if MATPLOTLIB_LT_21:
+        pytest.skip('Incompatible with old matplotlib')
+    
+    plt.rc('text', usetex=True)
+
+    header = {
+        'NAXIS': 2,
+        'NAXIS1': 360,
+        'NAXIS2': 180,
+        'CRPIX1': 180.5,
+        'CRPIX2': 90.5,
+        'CRVAL1': 180.0,
+        'CRVAL2': 0.0,
+        'CDELT1': -2 * np.sqrt(2) / np.pi,
+        'CDELT2': 2 * np.sqrt(2) / np.pi,
+        'CTYPE1': 'RA---MOL',
+        'CTYPE2': 'DEC--MOL',
+        'RADESYS': 'ICRS'}
+
+    wcs = WCS(header)
+    fig, ax = plt.subplots(
+        subplot_kw=dict(frame_class=EllipticalFrame, projection=wcs))
+    ax.set_xlim(-0.5, header['NAXIS1'] - 0.5)
+    ax.set_ylim(-0.5, header['NAXIS2'] - 0.5)
+    ax.coords[0].set_ticklabel(exclude_overlapping=True)
+    ax.coords[1].set_ticklabel(exclude_overlapping=True)
+    ax.coords[0].set_ticks(spacing=45 * u.deg)
+    ax.coords[1].set_ticks(spacing=30 * u.deg)
+    ax.grid()
+
+    if not matplotlib.checkdep_usetex(True):
```

Comment:
Why is this not at the top of the test function?

Reply:
Because I was flailing while trying to fix the skipif stuff.